doc/conf.py

```diff
@@ -89,7 +89,7 @@ def __getattr__(cls, name):
     "mlir_quantum.dialects.scf",
     "mlir_quantum.dialects.quantum",
     "mlir_quantum.dialects.gradient",
-    "mlir_quantum._mlir_libs._quantumDialects.quantum",
+    "mlir_quantum._mlir_libs._catalystDriver",
```

Comment:
I guess typically it indicates a "private" or "internal" module, but I think we want to expose the module "publicly" to the frontend. Are we able to place it at the top-level actually, so `mlir_quantum.catalyst_driver`, rather than nested inside  `_mlir_libs`?

Reply:
I have tried to rename the module but failed to do so quickly. My best attempt so far is [here](https://github.com/PennyLaneAI/catalyst/blob/9a4eee1acce3d16b5ed382a84c9c30c379b0f052/mlir/python/CMakeLists.txt#L54).

Reply:
It's probably because we use MLIR CMake functions that were designed for the dialect bindings for our driver.

I think there might be two options:
- add a Python module like those in `catalyst/mlir/python/dialects` that import the autogenerated module
- use our own CMake code to generate the Pybind11 module

Reply:
Hey, I moved it to `mlir_quantum.compiler_driver`. But we still have its name as `_catalystDriver`. Do you want to remove the underscore?

Reply:
Ideally yes. Does `compiler_driver` serve any purpose, or does it only contain the `_catalystDriver` module? If the latter it probably doesn't need to be there, wdyt?

Something that occurred to me as well is that this compiler driver might effectively become the new frontend of the Catalyst package, once the PL frontend component is moved into PL. But that's for the future.

Reply:
Changed name here: https://github.com/PennyLaneAI/catalyst/pull/216/commits/20e3d1644feed65860c0c5ff2ccc504cd3da1f83